addons/website/models/ir_rule.py

```diff
@@ -9,14 +9,15 @@ class IrRule(models.Model):
     @api.model
     def _eval_context(self):
         res = super(IrRule, self)._eval_context()
-
         # We need is_frontend to avoid showing website's company items in backend
         # (that could be different than current company). We can't use
         # `get_current_website(falback=False)` as it could also return a website
         # in backend (if domain set & match)..
         is_frontend = ir_http.get_request_website()
+        # Do not provide the current website in context if the rule is about websites.
+        is_not_on_website = self and self.model_id.model != 'website'
```

Comment:
Nitpicking but I'd go with
```suggestion
        is_website = self and self.model_id.model == 'website'
```
To avoid the double negation